iroha/src/block_sync.rs

```diff
@@ -115,41 +127,51 @@ pub mod message {
     #[derive(Io, Decode, Encode, Debug, Clone)]
     pub enum Message {
         /// Gossip about latest block.
-        LatestBlock((Hash, PeerId)),
+        LatestBlock(Hash, PeerId),
         /// Request for blocks after the block with `Hash` for the peer with `PeerId`.
-        GetBlocksAfter((Hash, PeerId)),
-        /// The response to `GetBlocksAfter`. Contains the requested blocks.
-        ShareBlocks(Vec<ValidBlock>),
+        GetBlocksAfter(Hash, PeerId),
+        /// The response to `GetBlocksAfter`. Contains the requested blocks and the id of the peer who shared them.
+        ShareBlocks(Vec<ValidBlock>, PeerId),
     }
 
     impl Message {
         /// Handles the incoming message.
         pub async fn handle(&self, block_sync: &mut BlockSynchronizer) {
             match self {
-                Message::LatestBlock((hash, peer)) => {
+                Message::LatestBlock(hash, peer) => {
                     let latest_block_hash = block_sync.kura.read().await.latest_block_hash();
                     if *hash != latest_block_hash {
                         if let Err(err) =
-                            Message::GetBlocksAfter((latest_block_hash, block_sync.peer_id.clone()))
+                            Message::GetBlocksAfter(latest_block_hash, block_sync.peer_id.clone())
                                 .send_to(peer)
                                 .await
                         {
                             eprintln!("Failed to request blocks: {:?}", err)
                         }
                     }
                 }
-                Message::GetBlocksAfter((hash, peer)) => {
-                    //TODO: send batches of configurable size instead of all blocks
-                    if let Some(blocks) = block_sync.kura.read().await.blocks_after(*hash) {
-                        if let Err(err) = Message::ShareBlocks(blocks.to_vec()).send_to(peer).await
-                        {
-                            eprintln!("Failed to send blocks: {:?}", err)
+                Message::GetBlocksAfter(hash, peer) => {
+                    if block_sync.batch_size > 0 {
```

Comment:
@eadventurous please add `eprintln!` if it is less or equal to zero

Reply:
It can't be less then zero as it is unsigned :smile: But yes, I will add a case for zero.